File: Java/opentraceability/src/main/java/opentraceability/interfaces/IVocabularyElement.java
package opentraceability.interfaces;

import opentraceability.utility.attributes.OpenTraceabilityJsonAttribute;
import org.json.JSONObject;

import java.util.ArrayList;

public abstract class IVocabularyElement {
    public abstract String getId();
    public abstract void setId(String val);

    public JSONObject context = null;

    public String epcisType = null;

    public VocabularyType vocabularyType = null;

    @OpenTraceabilityJsonAttribute(name = "@type")
    public String type;

    public ArrayList<IMasterDataKDE> kdes = new ArrayList<IMasterDataKDE>();

    public String getEpcisType()
    {
        return epcisType;
    }
}

File: Java/OpenTraceability/src/main/kotlin/opentraceability/interfaces/ivocabularyelement.kt
package interfaces

import org.json.JSONObject

interface IVocabularyElement {
    var ID: String?
    var EPCISType: String?
    var JsonLDType: String?
    var Context: JSONObject?
    var VocabularyType: VocabularyType
    var KDEs: ArrayList<IMasterDataKDE>
}
